netconf/rpc_test.go

```diff
@@ -216,18 +216,44 @@ configuration check-out failed: (missing mandatory statements)
 <name>fpc0</name>
 <commit-check-success/>
 </routing-engine>
-</commit-results>
 <ok/>
 </rpc-reply>`,
+		true,
+		false,
+	},
+	{
+		`
+<rpc-reply xmlns="urn:ietf:params:xml:ns:netconf:base:1.0"
+xmlns:junos="http://xml.juniper.net/junos/21.1R0/junos">
+<rpc-error>
+<error-severity>warning</error-severity>
+<source-daemon>
+rtlogd
+</source-daemon>
+<error-path>
+[edit security log]
+</error-path>
+<error-info>
+<bad-element>
+report
+</bad-element>
+</error-info>
+<error-message>
+It is recommended to use Stream Logging to an external logging server.
+</error-message>
+</rpc-error>
+<ok/>
+</rpc-reply>`,
+		false,
 		false,
 	},
 }
 
 func TestNewRPCReply(t *testing.T) {
 	for _, tc := range RPCReplytests {
 		reply, err := newRPCReply([]byte(tc.rawXML), false, "101")
-		if err != nil {
-			t.Fatalf("unexpected error: %v", err)
+		if (err == nil) == tc.Error {
```

Comment:
This can rather be simplified as:
```go
		if err != tc.Error {
```

Reply:
modified this to accomodate the new way of comparison.